performance_regression_test.py

```diff
@@ -185,6 +185,72 @@ def test_mixed(self):
         self.check_regression()
 
+    def test_latency(self):
+        """
+        Test steps:
+
+        1. Prepare cluster with data (reach steady_stet of compactions and ~x10 capacity than RAM.
+        with round_robin and list of stress_cmd - the data will load several times faster.
+        2. Run WRITE workload with gauss population.
+        """
+
+        # TO DO: add limit ops based on results.
+        prepare_write_cmd = self.params.get('prepare_write_cmd')
+        base_cmd_w = self.params.get('stress_cmd_w')
+        base_cmd_r = self.params.get('stress_cmd_r')
+        base_cmd_m = self.params.get('stress_cmd_m')
+
+        self.create_test_stats()
+
+        stress_queue = list()
+        # if test require a pre-population of data
+        if prepare_write_cmd:
+            params = {'prefix': 'preload-'}
+            # Check if the prepare_cmd is a list of commands
+            if not isinstance(prepare_write_cmd, basestring) and len(prepare_write_cmd) > 1:
+                # Check if it should be round_robin across loaders
+                if self.params.get('round_robin', default='').lower() == 'true':
+                    self.log.debug('Populating data using round_robin')
+                    params.update({'stress_num': 1, 'round_robin': True})
+
+                for stress_cmd in prepare_write_cmd:
+                    params.update({'stress_cmd': stress_cmd})
+
+                    # Run all stress commands
+                    self.log.debug('RUNNING stress cmd: {}'.format(stress_cmd))
+                    stress_queue.append(self.run_stress_thread(**params))
+
+            # One stress cmd command
+            else:
+                    stress_queue.append(self.run_stress_thread(stress_cmd=prepare_write_cmd, stress_num=1,
+                                                               prefix='preload-'))
+
+        for stress in stress_queue:
+            self.get_stress_results(queue=stress, store_results=False)
+
+        # Run WRITE workload
+        stress_queue = self.run_stress_thread(stress_cmd=base_cmd_w, stress_num=1)
+        results = self.get_stress_results(queue=stress_queue)
+        self.update_test_details()
+        # TEMP check if possible
+        self.display_results(results, test_name='test_latency')
+
+        # Run READ workload
+        self.create_test_stats()
+        stress_queue = self.run_stress_thread(stress_cmd=base_cmd_r, stress_num=1)
+        results = self.get_stress_results(queue=stress_queue)
+        self.update_test_details()
+        self.display_results(results, test_name='test_latency')
+
+        # run MIXED workload
+        self.create_test_stats()
+        stress_queue = self.run_stress_thread(stress_cmd=base_cmd_m, stress_num=1)
+        results = self.get_stress_results(queue=stress_queue)
+        self.update_test_details()
+        self.display_results(results, test_name='test_latency')
+
+        self.check_regression()
```

Comment:
this method should be called for each workload separately

Reply:
Fixed.